src/Features/Core/Portable/ExternalAccess/VSTypeScript/VSTypeScriptNavigateToSearchService.cs

```diff
@@ -3,133 +3,127 @@
 
-        public IImmutableSet<string> KindsProvided => _searchService?.KindsProvided ?? ImmutableHashSet<string>.Empty;
+    public IImmutableSet<string> KindsProvided => _searchService?.KindsProvided ?? ImmutableHashSet<string>.Empty;
 
-        public bool CanFilter => _searchService?.CanFilter ?? false;
+    public bool CanFilter => _searchService?.CanFilter ?? false;
 
-        public async Task SearchDocumentAsync(
-            Document document,
-            string searchPattern,
-            IImmutableSet<string> kinds,
-            Document? activeDocument,
-            Func<INavigateToSearchResult, Task> onResultFound,
-            CancellationToken cancellationToken)
+    public async Task SearchDocumentAsync(
+        Document document,
+        string searchPattern,
+        IImmutableSet<string> kinds,
+        Func<INavigateToSearchResult, Task> onResultFound,
+        CancellationToken cancellationToken)
+    {
+        if (_searchService != null)
         {
-            if (_searchService != null)
-            {
-                var results = await _searchService.SearchDocumentAsync(document, searchPattern, kinds, cancellationToken).ConfigureAwait(false);
-                foreach (var result in results)
-                    await onResultFound(Convert(result)).ConfigureAwait(false);
-            }
+            var results = await _searchService.SearchDocumentAsync(document, searchPattern, kinds, cancellationToken).ConfigureAwait(false);
+            foreach (var result in results)
+                await onResultFound(Convert(result)).ConfigureAwait(false);
         }
+    }
 
-        public async Task SearchProjectAsync(
-            Project project,
-            ImmutableArray<Document> priorityDocuments,
-            string searchPattern,
-            IImmutableSet<string> kinds,
-            Document? activeDocument,
-            Func<INavigateToSearchResult, Task> onResultFound,
-            CancellationToken cancellationToken)
-        {
-            if (_searchService != null)
-            {
-                var results = await _searchService.SearchProjectAsync(project, priorityDocuments, searchPattern, kinds, cancellationToken).ConfigureAwait(false);
-                foreach (var result in results)
-                    await onResultFound(Convert(result)).ConfigureAwait(false);
-            }
-        }
+    public async Task SearchProjectsAsync(
+        Solution solution,
+        ImmutableArray<Project> projects,
+        ImmutableArray<Document> priorityDocuments,
+        string searchPattern,
+        IImmutableSet<string> kinds,
+        Document? activeDocument,
+        Func<Project, INavigateToSearchResult, Task> onResultFound,
+        Func<Task> onProjectCompleted,
+        CancellationToken cancellationToken)
+    {
+        using var _ = PooledHashSet<Project>.GetInstance(out var processedProjects);
 
-        public Task SearchCachedDocumentsAsync(
```

Comment:
these became optional methods.  so TS/F# no longer need to have dummy impls of them.